source/hic_hal/stm32/stm32h743ii/extended_features/include/adapter_adc.h

```diff
@@ -0,0 +1,13 @@
+#ifndef _ADAPTER_ADC_H_INCLUDED_
```

Comment:
I think we need to separate the adc and the adapter_detector further.

The ADC should be initialized as part of each ADC read, because we might use the adc for other purposes in between. The adc doesn't strictly belong to the adapter detector. Epidote has something similar, you can take a look at nlnq_accessory.c

At nest, we also like to prefix our global functions with the file name. So if this file becomes adc_detector, the function names would be like adc_detector_update()

Reply:
I think that's not a problem. If there's no channel group supported by this chip, then I think it should be no problem to have your implementation ignore the channelGroup parameter.